src/Mailer/Renderer.php

```diff
@@ -0,0 +1,462 @@
+<?php
+declare(strict_types=1);
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Mailer;
+
+use Cake\Core\Configure;
+use Cake\Http\Client\FormDataPart;
+use Cake\Utility\Hash;
+use Cake\Utility\Security;
+use Cake\Utility\Text;
+
+class Renderer
```

Comment:
This should have a class doc. Is this class a public one that folks can extend/replace/rely on or is it intended to be an internal one?

Reply:
We could make it internal. I personally can't think of a use case where someone might want to extend/modify it.